tests/roundtrippable_types.tests

```diff
@@ -12,7 +12,10 @@ filemode : true
 stdout : (q = [3, 1, 2]) : Q
 
 Recursive effect
-tests/roundtrip/recursive_effect.links
-filemode : true
+fun w(g) { handle(g()) { case Return(()) -> () case F(h, res) -> w(h) }} w
 args : --enable-handlers
-ignore : not roundtrippable
\ No newline at end of file
+stdout : fun : (() { |(mu a . F:(() { |a}~> ()) {}-> _::Any,wild:()|c)}~> ()) {F{_}|c}~> ()
+
+Recursive variant
+sig f : ([|(mu a. Op:() { |a}-> ())|]) {}-> () fun f(_) { () } f
+stdout : fun : ([| |(mu a . Op:() { |a}-> ())|]) {}-> ()
```

Comment:
I don't think this type is entirely correct. The `|` in front of `(mu ...` should not be printed here, I think.

Reply:
It seems internally the `(mu ...)` is in a row variable (at least from the non-pretty printout)
```ocaml
(Types.Function
   ((Types.Record (Types.Row
                     ({1 => (Types.Present
                               (Types.Variant
                                  (Types.Row
                                     ({}, Point @ (...) (Types.Recursive (6642,...

```

Reply:
This is now resolved, variants have special treatment - no pipe for row variable.